### PR TITLE
WCOR-1228: Follow-up to refine faceted group documents list.

### PR DESCRIPTION
**Pull request by lgbridges**
### Use case

As an admin, I want this view to apply to all media and not just documents. I also want Media to be indexed separately from Content.

As a user, some refinement of how the view-pages are displayed and how facets work is warranted.

### Implementation Details

Revision of #271. Duplicate documents_solr view to media_solr, and create an all Media page (/group/media) while retaining the Documents-only page (/group/documents). Create separate Media index to be used by this view. Adjust facets and list display to ensure hierarchical term groupings display as expected.

### Test Plan

- [x] Prepare a set of content to match an example such as [Finance Reports](https://www.co.washington.or.us/Support_Services/Finance/CountyFinancials/financial-reports.cfm) from Finance group. Matching the example page, begin by adding the term "Financial Reports" to the Tags taxonomy, and then add the term "Component Units" as a child term to the Financial Reports term.
- [x] Create 4 test documents using any dummy PDF file. All documents should have Displayed In value of "Finance".
   - One document should be named Comprehensive Annual Financial Report, have Financial Reports as a Tags value, and Fiscal Year 2020.
   - The next document should be the same as above, but with Fiscal Year 2019.
   - The next document should be named Enhanced Sheriff Patrol District, have Component Units as a Tags value, and Fiscal Year 2020. 
   - The next document should be the same as above, but with Fiscal Year 2019.
- [x] Create 2 test video/audio items using any YouTube link. All items should have Displayed In value of "Finance".
   - One video/audio item should be named Comprehensive Annual Financial Summary Presentation, have Financial Reports as a Tags value, and Fiscal Year 2020.
   - The next video/audio item should be the same as above, but with Fiscal Year 2019.
- [x] Rebuild media index. (Visit path /admin/config/search/search-api/index/acquia_search_index_media, click red _Queue all items for reindexing_ button, confirm the action, then under _Start indexing now_ section click the _Index now_ button.
- [x] Visit path /finance/media to see a list view of the media (both documents and a/v items). Note that when you filter by facet, a new URL is generated that can be used to present a specifically filtered list of documents (used to create a Resource for a persistent view, etc).
- [x] Visit path /finance/documents to see a list view of documents only. Video/audio items will not be displayed.

### CODE REVIEW

**Comment by lgbridges** [comment]
Minor adjustment to test plan.

**Comment by jmilden-git** [comment]
@lgbridges For some reason the admin user on ODE230 is not able to upload files.  This is the resulting error msg:
Upload error. Could not move uploaded file Washington-County-2020-SEFA.pdf to destination s3://media/document/upload/2021/Washington-County-2020-SEFA.pdf.
How should I proceed with QA?

**Comment by lgbridges** [comment]
> @lgbridges For some reason the admin user on ODE230 is not able to upload files. This is the resulting error msg: Upload error. Could not move uploaded file Washington-County-2020-SEFA.pdf to destination s3://media/document/upload/2021/Washington-County-2020-SEFA.pdf. How should I proceed with QA?

@jmilden-git -- Going to go ahead and refactor after workshopping this with Josh a bit more, so QA is on pause (sorry). 

The issue you mention has to do with the S3 secrets not making it to the CD environment being used for QA. I think Jung set it up so that the first login to the ODE using BLT should upload the secrets, but occasionally they don't make it. If you run `blt custom:secret 230` from your IDE, BLT should upload the secrets manually to this ODE and once you refresh the page / flush cache, hopefully the document upload will then work.

For your back pocket when this comes back to QA, thanks.

**Comment by jmilden-git** [comment]
> The issue you mention has to do with the S3 secrets not making it to the CD environment being used for QA. I think Jung set it up so that the first login to the ODE using BLT should upload the secrets, but occasionally they don't make it. If you run `blt custom:secret 230` from your IDE, BLT should upload the secrets manually to this ODE and once you refresh the page / flush cache, hopefully the document upload will then work.
> 
> For your back pocket when this comes back to QA, thanks.

That fixed it.  I vaguely recall there was a command.  Thanks for the reminder.  Let me know when it's ready to resume.

**Comment by lgbridges** [comment]
Awaiting required webform module update. See #292.

**Comment by jmilden-git** [comment]
@lgbridges The second to last QA step resulted in an error msg on screen.  Otherwise the documents and videos were there like expected.  Also, the step to add fiscal year tags had to be configured first.  I added to FY2019 / FY2020 tags so I could complete the step.
image

File: docroot/themes/custom/westy/templates/media/media--video--embedded.html.twig [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+#}
+{%
+  set classes = [
+    'embed-video',
+    'embedded-link'
+  ]
+%}
+
+<div{{ attributes.addClass(classes) }}>
```
Comment (washco-or) [inline docroot/themes/custom/westy/templates/media/media--video--embedded.html.twig:41]: @lgbridges curious as to why this file was created.

**Comment by jmilden-git** [comment]
QA has passed.

**Review by jmilden-git — APPROVED** [review]

**Pull request merged**